packages/core/src/component.ts

```diff
@@ -27,13 +31,21 @@ export interface Component {
    * A map of name/class pairs for binding providers
    */
   providers?: ProviderMap;
+
+  classes?: ClassMap;
```

Comment:
I find this name confusing. What does it mean for a component to export a class? Some classes are exported via JavaScript/TypeScript exports, e.g. `export class FooBar`. Why should be some classes exported via JS/TS and some others exported via Component properties?

Personally, I prefer to remove this API entirely, keep `providers` for backwards compatibility (possibly even deprecate it) and ask users to leverage `providers` API for all bindings.

If you feel it's important to keep this short-cut, then let's find a better name please. For example:`classBindings` or `classesToBind`.

Reply:
Oops, I made a typo in my comment above.

```diff
- ask users to leverage `providers` API for all bindings.
+ ask users to leverage `bindings` API for all bindings.
```

Reply:
I added jsdocs to make it clear.

Reply:
I am afraid jsdocs are not good enough. Consider the following code snippet from your pull request:

```ts
      class MyComponentWithClasses implements Component {
         classes = {'my-class': MyClass};
       }
```

The person reading the code (e.g. while reviewing a pull request on GitHub) does not see tscode comments.